### PR TITLE
PCA: Use it as feature scorer

### PR DESCRIPTION
**Pull request by VesnaT**
Add an output to PCA widget that can go to Rank widget.

### CODE REVIEW

**Comment by BlazZupan** [comment]
Other scorers (like linear and logistic regression) provide scoring method. PCA scores features only when given data (I've tried this on widgets). There's some redundancy in this: Rank widget is given the data, and so is PCA as well. Was this intentional?

**Comment by lanzagar** [comment]
The scorer that is sent as the output from the PCA widget does not perform normalization. Even if that option was selected in the widget.
Especially for scoring features, normalization should almost always be used!

File: Orange/widgets/unsupervised/owpca.py [hunk 1]
```diff
@@ -147,13 +162,13 @@ def start(self):
         else:
             self.address = self.addresstext.text()
             with remote.server(self.address):
-                from Orange.projection.pca import RemotePCA
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owpca.py]: RemotePCA needs to be imported inside the remote context!

**Comment by janezd** [comment]
@lanzagar, can you check and merge if this is OK now?

File: Orange/projection/pca.py [hunk 2]
```diff
@@ -10,14 +11,35 @@ def save_state(_):
...
 from Orange.misc.wrapper_meta import WrapperMeta
 from Orange.preprocess import Continuize
 from Orange.projection import SklProjector, Projection
+from Orange.preprocess.score import LearnerScorer, LearnerScorerError
 
 __all__ = ["PCA", "SparsePCA", "RandomizedPCA", "IncrementalPCA"]
 
 
-class PCA(SklProjector):
+class PCAComponentOutOfRangeError(LearnerScorerError):
```
Comment (lanzagar) [inline Orange/projection/pca.py]: This Error in no longer needed?

File: Orange/tests/test_pca.py [hunk 3]
```diff
@@ -106,3 +106,26 @@ def test_chain(self):
...
+        data = Orange.data.Table('iris')
+        pca = PCA()
+        scores = pca.score_data(data)
+        self.assertEqual(len(scores), len(data.domain.attributes))
+        self.assertGreater(sum(scores), 0)
+        self.assertEqual(['petal length', 'sepal length'],
+                         sorted([data.domain.attributes[i].name
+                                 for i in np.argsort(scores)[-2:]]))
+
+    def test_PCA_scorer_component(self):
```
Comment (lanzagar) [inline Orange/tests/test_pca.py]: The test should now probably check the first component.

File: Orange/tests/test_pca.py [hunk 4]
```diff
@@ -106,3 +106,26 @@ def test_chain(self):
...
+        data = Orange.data.Table('iris')
+        pca = PCA()
+        pca.component = 1
+        scores = pca.score_data(data)
+        self.assertEqual([round(s, 4) for s in scores],
+                         [0.6565, 0.7297, 0.1758, 0.0747])
+
+    def test_PCA_scorer_component_out_of_range(self):
```
Comment (lanzagar) [inline Orange/tests/test_pca.py]: This test in no longer relevant.

**Pull request merged**